kubernetes/metadata/resource.go

```diff
@@ -113,8 +123,18 @@ func (r *Resource) GenerateK8s(kind string, obj kubernetes.Resource, options ...
 		},
 	}
 
-	if accessor.GetNamespace() != "" {
-		_ = safemapstr.Put(meta, "namespace", accessor.GetNamespace())
+	namespaceName := accessor.GetNamespace()
+	if namespaceName != "" {
+		_ = safemapstr.Put(meta, "namespace", namespaceName)
+
+		if r.namespace != nil {
+			nsMeta := r.namespace.GenerateFromName(namespaceName)
+			if nsMeta != nil {
+				// Use this in 8.0
+				//out.Put("namespace", meta["namespace"])
```

Comment:
Most probably this is not needed. We have concluded that we are not changing anything here. You can remove  completely  this comment :)